fixed-main-height-calculation

djidji01 opened the pull request:
here is my proposal! #3150 , #3137 and #3123

danny007in reviewed (commented):
See the bug in top-nav.html file

![Screenshot 2020-11-26 172344](https://user-images.githubusercontent.com/50356015/100348022-4c227700-300c-11eb-873f-5fe51fda5256.gif)

djidji01 commented:
> See the bug in top-nav.html file,
> its still remains same over there

this is due to the fact that is missplaced title for `.control-sidebar` outside `.control-sidebar-content` . moving it inside solve the issue.

![image](https://user-images.githubusercontent.com/19886674/100352692-0eb2ef00-2ff6-11eb-9aab-aa610a80bb52.png)

danny007in commented:
> 
> 
> > See the bug in top-nav.html file,
> > its still remains same over there
> 
> this is due to the fact that is missplaced title for `.control-sidebar` outside `.control-sidebar-content` . moving it inside solve the issue.
> 
> ![image](https://user-images.githubusercontent.com/19886674/100352692-0eb2ef00-2ff6-11eb-9aab-aa610a80bb52.png)

OK ill trying to fix

danny007in commented on build/js/Layout.js:
```diff
@@ -83,19 +82,29 @@ class Layout {
       }
 
       if (this._isFooterFixed()) {
-        $contentSelector.css(this._config.panelAutoHeightMode, parseFloat($contentSelector.css(this._config.panelAutoHeightMode)) + heights.footer)
```
both are same

REJack commented:
The calculation was fixed with @danny007in's PR #3158, I see need for this PR. 
I've removed the unnecessary control-sidebar demo content.

djidji01 commented:
> The calculation was fixed with @danny007in's PR #3158, I see need for this PR.

It was not fixed at all My Dear!! 

to prove it just try and remove a single link item from sidebar, toggle the control-sidebar and see the same bug. the behavior you see with   PR #3158 is because of the new **_Tabbed IFrame Plugin_** that make the `($('.sidebar').height() - $('.control-sidebar-content').height()) > $('.footer').height()`

REJack commented:
I see what you mean, I've added a fix for this in master repo please test it 😄.

djidji01 commented:
> I see what you mean, I've added a fix for this in master repo please test it

Yeap fixed, however: 

- It seems @danny007in commit is not applied. footer is still not at the bottom

- Shouldn't the  `heights.controlSidebar + heights.footer >= heights.sidebar && heights.controlSidebar != 0` check replace `max === heights.controlSidebar`  just to not repeat the same thing.

- shouldn't the `$body.hasClass(CLASS_NAME_LAYOUT_TOP_NAV)` be removed because my cause unexpected behavior since the bag was only due to the misplaced` title content`.

regards

danny007in reviewed (commented):
> 
> 
> > I see what you mean, I've added a fix for this in master repo please test it
> 
> Yeap fixed, however:
> 
>     * It seems @danny007in commit is not applied. footer is still not at the bottom
> 
>     * Shouldn't the  `heights.controlSidebar + heights.footer >= heights.sidebar && heights.controlSidebar != 0` check replace `max === heights.controlSidebar`  just to not repeat the same thing.
> 
>     * shouldn't the `$body.hasClass(CLASS_NAME_LAYOUT_TOP_NAV)` be removed because my cause unexpected behavior since the bag was only due to the misplaced` title content`.
> 
> 
> regards



> 
> 
> > I see what you mean, I've added a fix for this in master repo please test it
> 
> Yeap fixed, however:
> 
>     * It seems @danny007in commit is not applied. footer is still not at the bottom
> 
>     * Shouldn't the  `heights.controlSidebar + heights.footer >= heights.sidebar && heights.controlSidebar != 0` check replace `max === heights.controlSidebar`  just to not repeat the same thing.
> 
>     * shouldn't the `$body.hasClass(CLASS_NAME_LAYOUT_TOP_NAV)` be removed because my cause unexpected behavior since the bag was only due to the misplaced` title content`.
> 
> 
> regards

djidji01 commented:
> I see what you mean, I've added a fix for this in master repo please test it 😄.



> I see what you mean, I've added a fix for this in master repo please test it 😄.

Also the code will cause unexpected view on.  `. content-wrapper ` when  `max===window. height() `

djidji01 commented on build/js/Layout.js:
```diff
@@ -83,19 +82,29 @@ class Layout {
         } else {
-          $contentSelector.css(this._config.panelAutoHeightMode, (max + offset))
+          $contentSelector.css(this._config.panelAutoHeightMode, (max + offset) - heights.footer)
         }
-      } else if (max === heights.window) {
-        $contentSelector.css(this._config.panelAutoHeightMode, (max + offset) - heights.header - heights.footer)
       } else {
         $contentSelector.css(this._config.panelAutoHeightMode, (max + offset) - heights.header)
       }
 
       if (this._isFooterFixed()) {
-        $contentSelector.css(this._config.panelAutoHeightMode, parseFloat($contentSelector.css(this._config.panelAutoHeightMode)) + heights.footer)
+        $contentSelector.css(this._config.panelAutoHeightMode, parseFloat($contentSelector.css('min-height')) + heights.footer)
```
Sure.

REJack commented:
@djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.

djidji01 commented:
> 
> 
> @djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.

@REJack  As I said just look what happen when window is the max height!!!

![image](https://user-images.githubusercontent.com/19886674/100450760-e2af7080-30be-11eb-91a9-4e5b3e846ff4.png)

danny007in commented:
> > 
> > 
> > @djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.
> 
> @REJack  As I said just look what happen when window is the max height!!!
> 
> ![image](https://user-images.githubusercontent.com/19886674/100450760-e2af7080-30be-11eb-91a9-4e5b3e846ff4.png)
> 
> 

Actual compilation master branch will be before new release.

REJack commented:
Does this bug still exist in the master branch?

danny007in commented:
> 
> 
> Does this bug still exist in the master branch?

No,
Close it

djidji01 commented:
> 
> 
> > @djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.
> 
> @REJack As I said just look what happen when window is the max height!!!
> 
> ![image](https://user-images.githubusercontent.com/19886674/100450760-e2af7080-30be-11eb-91a9-4e5b3e846ff4.png)

@danny007in @REJack unless you neglected this and it is still a bug.

thanks

danny007in commented:
> 
> 
> > > @djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.
> > 
> > 
> > @REJack As I said just look what happen when window is the max height!!!
> > ![image](https://user-images.githubusercontent.com/19886674/100450760-e2af7080-30be-11eb-91a9-4e5b3e846ff4.png)
> 
> @danny007in @REJack unless you neglected this and it is still a bug.
> 
> thanks

@djidji01 Merge master again and Check now 😄

danny007in commented:
> 
> 
> > > @djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.
> > 
> > 
> > @REJack As I said just look what happen when window is the max height!!!
> > ![image](https://user-images.githubusercontent.com/19886674/100450760-e2af7080-30be-11eb-91a9-4e5b3e846ff4.png)
> 
> @danny007in @REJack unless you neglected this and it is still a bug.
> 
> thanks

OK i checked this, Yes its not fixed @djidji01 @REJack

danny007in commented:
@REJack Not solved 

![8U1G35n65J](https://user-images.githubusercontent.com/50356015/101774666-6bdd9300-3b14-11eb-8859-7f1947251c0f.gif)
